falcon/request.py

```diff
@@ -256,6 +257,9 @@ def __init__(self, env, options=None):
         self.stream = env['wsgi.input']
         self.method = env['REQUEST_METHOD']
 
+        if self.stream.__class__.__name__ == InputWrapper.__name__:
```

Comment:
@kgriffs I just confirmed, an instance of `Request` isn't created during testing until some time after `wsgiref.validate.validator()` is called ([L327](https://github.com/falconry/falcon/blob/master/falcon/testing/test_case.py#L327) in `testing/test_case.py`). Since L327 is the one that errors out during a `simulate_request()` there'd be no chance for that wrap check in L306 to work its magic.